### PR TITLE
feat: untangle the async thorchain deposit madness

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- tackles this guy, leveraging query caching to make us faster and less overfetchy

https://github.com/shapeshift/web/blob/e5451a91186b77b2541856124fc5522fa4933327/src/lib/utils/thorchain/hooks/useGetThorchainSaversDepositQuoteQuery.tsx#L30

- uses learnings from react-query type-safety to make the THORChain deposit query type-safe
- untangles the IIAFE unsafe and unpredictable deposit madness of THORChain deposit at input and confirm step
- now that we do proper clean data fetching, leverages loading states to introduce skeletons for confirm step values

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

Low, see testing

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Do a quick smoke test around savers and confirm things are still happy e2e
- Slippage, time to break even and protocol fees are still happy

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

https://jam.dev/c/afe42d48-1325-4670-9d66-14d4b2816192

### CODE REVIEW

File: src/lib/utils/thorchain/hooks/useGetThorchainSaversDepositQuoteQuery.tsx [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+  amountCryptoBaseUnit: BigNumber.Value | null | undefined
+}
+
 export type GetThorchainSaversDepositQuoteQueryKey = [
   'thorchainSaversDepositQuote',
-  {
-    asset: Asset
-    amountCryptoBaseUnit: BigNumber.Value | null | undefined
-  },
+  PartialFields<FetchThorchainDepositQuoteInput, 'asset'>,
```
Comment (0xApotheosis) [inline src/lib/utils/thorchain/hooks/useGetThorchainSaversDepositQuoteQuery.tsx:16]: This seems to work as expected, but should we have `amountCryptoBaseUnit` also a part of the query key?

**Review by 0xApotheosis — APPROVED** [review]
Smoke test passed, savers is looking happy.

File: src/lib/utils/thorchain/hooks/useGetThorchainSaversDepositQuoteQuery.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/lib/utils/thorchain/hooks/useGetThorchainSaversDepositQuoteQuery.tsx:16]: We do! `PartialFields` makes some fields as optional, while keeping others as they were (optional or required) so this is effectively 
```typescript
type FetchThorchainDepositQuoteInput = {
  asset: Asset
  amountCryptoBaseUnit?: BigNumber.Value | null | undefined
}
```

Comment (0xApotheosis) [reply src/lib/utils/thorchain/hooks/useGetThorchainSaversDepositQuoteQuery.tsx:16]: Oh, right you are ser!

**Pull request merged**